docs: Update deciding-about-your-stack.md

crobbins215 opened the pull request:
## Description

The first part of updating the greenfield documentation for https://github.com/camunda/product-hub/issues/2146, the architectural diagram hasn't been updated yet. Changes are wording changes and removing documentation that arent necessary. 

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->
<!-- Help the DevEx team prioritize our work (reviews, merges, etc.) by opening PRs sooner. -->

- [x] This change or page is the start of an ongoing effort to improve the greenfield stack documentation.
- [x] This functionality is already available and is a update to the current doc.


## PR Checklist

<!-- Keep in mind, Camunda maintains 18 months of versions. Backporting your change or including it in multiple versions is common. -->

- [x] I have added changes to the main `/docs` directory (aka `/next/`)
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @akeller as a reviewer

akeller commented:
I had some time to run prettier to correct the formatting issues and get those out of the way.

I also noticed some style guide issues - we use sentence case for our titles. You can find the cheatsheet here - https://github.com/camunda/camunda-docs/blob/main/howtos/technical-writing-cheatsheet.md

crobbins215 commented on docs/components/best-practices/architecture/deciding-about-your-stack.md:
```diff
@@ -9,83 +9,69 @@
 
-1. Model the processes with the [Camunda Modeler](https://camunda.org/download/modeler/).
+2. **Spring Boot:**
+   1. **Develop a Spring Boot Application:** Users should develop their process solutions as [Spring Boot](https://spring.io/projects/spring-boot) applications. This involves setting up a new Spring Boot project, either manually or using tools like [Spring Initializr](https://start.spring.io/).
```
Accepted

crobbins215 commented on docs/components/best-practices/architecture/deciding-about-your-stack.md:
```diff
@@ -9,83 +9,69 @@
 
-To run the process application _in production_:
+4. **IDE Selection:**
+   1. **Choose an IDE:** Select an Integrated Development Environment (IDE) that supports Java development, Maven, and Spring Boot. Frequently used options include Visual Studio Code, IntelliJ IDEA, or Eclipse.
```
Accpeted

crobbins215 commented:
@akeller, I've changed the links and accepted the prettier changes. I also changed the case type; are you able to double-check, as I wasn't 100% on all of them?

akeller commented on docs/components/best-practices/architecture/deciding-about-your-stack.md:
```diff
@@ -9,83 +9,77 @@ tags:
 
-1. Use your favorite IDE, for example Visual Studio Code, IntelliJ or Eclipse.
+   1. **Develop a Spring Boot application:** Users should develop their process solutions as [Spring Boot](https://spring.io/projects/spring-boot) applications. This involves setting up a new Spring Boot project, either manually or using tools like [Spring Initializr](https://start.spring.io/).
+   2. **Integrate Spring Zeebe:** Integrate [Spring Zeebe](https://github.com/camunda-community-hub/spring-zeebe) into the Spring Boot project by adding necessary dependencies to the project’s pom.xml file and configuring the application to use Camunda services.
```
We have a lot of disclaimers around community clients since Spring Zeebe is not yet fully supported. We should add something here as well.

crobbins215 replied:
Yeah, that's a good add; we can remove that when it becomes part of the Camunda official offering.

akeller commented on docs/components/best-practices/architecture/deciding-about-your-stack.md:
```diff
@@ -9,83 +9,77 @@ tags:
-We like to give one greenfield stack recommendation, which is the stack you can simply use if there is no reason against it. And while we went through long and detailed discussions to come to this recommendation, it _doesn't_ mean that it is necessarily superior to alternative stacks. You can still feel confident if you go down another route (note below for alternative options).
-
-The stack looks pretty similar in various programming language. Please use the programming language your team is most familiar with. If in doubt, use Java or JavaScript.
+Our greenfield stack recommendation is a result of extensive discussions and evaluations. While not the only option, it is a solid choice if there are no specific reasons to choose an alternative. Your choice of programming language should align with your team's expertise; we suggest Java or JavaScript for their broad applicability and support.
```
We mention Java or Javascript, but then only cover Java in your changes. Is that intentional? I would remove Javascript if we won't talk about it.

crobbins215 replied:
How you have reworded the change works, lets mention it but point out this stack is a Java greenfield stack.

akeller commented on docs/components/best-practices/architecture/deciding-about-your-stack.md:
```diff
@@ -9,83 +9,77 @@ tags:
 ![greenfield stack architecture diagram](deciding-about-your-stack-assets/greenfield-architecture.png)
 
-Use the following stack:
+For a Java-based setup using Camunda 8 SaaS and Spring Boot, use the following stack:
```
This only covers SaaS, is that intentional? I would rework the intro sentences to set expectations that this only covers SaaS.

crobbins215 replied:
Currently yes, its focus is on SaaS while the SM changes happen, need to figure out how they are adjusted when that is done.

akeller replied:
Made a small change to the intro sentence.

akeller commented on docs/components/best-practices/architecture/deciding-about-your-stack.md:
```diff
@@ -9,83 +9,77 @@ tags:
 
-1. Use [Camunda 8 SaaS](https://signup.camunda.com/accounts?utm_source=docs.camunda.io&utm_medium=referral) and create a cluster there
+1. **Camunda 8 SaaS:**
 
-1. Build your process solution project as a [Spring Boot](https://spring.io/projects/spring-boot) application, using the [Spring Zeebe](/docs/apis-tools/community-clients/spring.md).
+   1. **Sign up:** Users need to sign up for Camunda 8 SaaS. This can be done through the [Camunda SaaS registration page.](https://signup.camunda.com/accounts?utm_source=docs.camunda.io&utm_medium=referral)
+   2. **Create a cluster:** After signing up, users should create a cluster. They can follow the guide on [Creating a Cluster in Camunda 8](/guides/create-cluster.md) which provides step-by-step instructions on setting up a new cluster in the Camunda 8 environment.
```
Since this is a Best Practices page, could we assume users already have an account and a cluster? I would like to drive everyone through our [Getting Started Guide content](https://docs.camunda.io/docs/guides/#getting-started) first.

crobbins215 replied:
Something more like?

1) If you're new to Camunda SaaS, please check out our Getting Started Guide to set up your environment.

2. **Create a cluster:** After signing up, users should create a cluster. They can follow the guide on [Creating a Cluster in Camunda 8](/guides/create-cluster.md) which provides step-by-step instructions on setting up a new cluster in the Camunda 8 environment.

akeller commented on docs/components/best-practices/architecture/deciding-about-your-stack.md:
```diff
@@ -9,83 +9,77 @@ tags:
-- SaaS is the easiest way to consume capabilities like a workflow engine.
-- Spring Boot is currently the most adopted way of building Java applications.
-- Spring Boot applications are easy to customize as well as easy to roll out into test and production environments, either on-premise or in the cloud.
+   1. **Use Camunda desktop Modeler:** Download and use the Camunda Modeler for designing and modeling business processes. The Modeler is available here.
```
The rest of this page is SaaS-first - why not Web Modeler?

crobbins215 replied:
This is deliberate, the greenfield stack will expand to cover self managed, also customers looking to implement using a spring boot application currently also align better with the desktop modeler for now. We wanted to avoid providing multiple choices at this stage but plan on adjusting as the document grows

Here link is missing, ill update that

akeller commented on docs/components/best-practices/architecture/deciding-about-your-stack.md:
```diff
@@ -9,83 +9,77 @@ tags:
-| Roundtrip in between process analysts and developers possible | &#10004;                                                                    | &#10004; (Carefully check level of BPMN compliance - the [Model Interchange Working Group](http://bpmn-miwg.github.io/bpmn-miwg-tools/) can serve as a first starting point |                                                 |
-| Use for process analysts                                      | &#10004;                                                                    | &#10004;                                                                                                                                                                    |                                                 |
-| Use for process developers                                    | &#10004;                                                                    |                                                                                                                                                                             |                                                 |
-| Use when                                                      | You do not have a BPMN standard compliant modeling tool already rolled out. | You already rolled out a BPMN tool with a standard compliancy sufficient for roundtrip.                                                                                     | Try to avoid                                    |
-|                                                               | [Download](https://camunda.org/download/modeler/)                           | [e.g. Cawemo](http://cawemo.com/)                                                                                                                                           |                                                 |
+Find more information in the [Camunda Platform 8 Documentation](https://docs.camunda.io/), or join the [Camunda Platform 8 Forum](https://forum.camunda.io/) and [Camunda Platform 8 Slack](https://camunda-cloud.slack.com/) community.
```
This section seems redundant, and our Slack doesn't exist anymore. We can link to the contact page, but we are already in the docs.

crobbins215 replied:
Okay, happy for removal

akeller commented:
@crobbins215 I pushed a commit with some formatting and content architecture adjustments for brevity and flow. I would read through the review comments I left first then look at the changes I made. In some cases, we may need to add or remove content based on discussion in the review comments.

Some notable changes in the commit -

* I removed the opening sentences. They seemed redundant. 
* I added a disclaimer for Spring Zeebe as a community supported project
* I added an HTML comment to add a description of the architecture in the image (thing of it as the TL;DR of the stack rather than just an image description)
* I moved "Why this stack?" up before the set up section. Sell the stack before someone goes through the effort of setting it up
* I reformatted the numerical list and bolding because I found it hard to read. The H4s also give us the ability to link. I don't love either format, but I do think removing the bolded text makes it easier to read and is less redundant.
* In the Polyglot Stacks section I added ", including Javascript". (I just realized these titles aren't sentence case, will fix in a coming commit.)
* I removed the "Further References" section

crobbins215 commented:
@akeller, 

I've added some comments on some of the suggestions above, which we can discuss in them areas; the doc is looking much better and more streamlined.  

The architectural diagram I'm working with design to see if they can come up with something improved, a description can be added then, I saw your code comment.

akeller commented:
I adjusted some of the sentences to be more direct. Instead of talking to "the users," let's talk to the reader 🙂 

@crobbins215 we usually [backport](https://github.com/camunda/camunda-docs/pull/3450#issuecomment-1988234378) our changes. Currently this is only in our `/next/` version, so it would be available in 8.5. Would you like to backport to all supported versions of Camunda 8? 

There are two HTML comments I'll call out with a review comment. I'll also hand this over to @christinaausley for a review to catch any style guide/tense/voice issues I may have missed.

akeller commented on docs/components/best-practices/architecture/deciding-about-your-stack.md:
```diff
@@ -9,83 +9,84 @@ tags:
 ![greenfield stack architecture diagram](deciding-about-your-stack-assets/greenfield-architecture.png)
 
-Use the following stack:
+<!-- Add a brief description of the architecture in the image -->
```
While the rest of this document does describe this architecture, can we add a brief 1-2 sentences about the image itself?

crobbins215 replied:
Something like
> This architecture depicts a user's browser request flow through Camunda SaaS, orchestrating workflows, to a Spring Boot application, managing business logic and database interactions.

crobbins215 replied:
Added a description

akeller commented on docs/components/best-practices/architecture/deciding-about-your-stack.md:
```diff
@@ -9,83 +9,84 @@ tags:
+
+For a practical implementation, refer to our [example application on GitHub](https://github.com/camunda-community-hub/camunda-cloud-examples/tree/main/twitter-review-java-springboot), which demonstrates a typical setup for a Spring Boot-based process application with Camunda.
+
+<!-- This should be moved from the Community Hub -->
```
This example sits on the Community Hub. We can follow up in a future PR, but I want to make sure we acknowledge this shouldn't be referenced since it may not be maintained.

crobbins215 replied:
Yeah, I've noted down that the example needs to be reworked in the future, as a Twitter example isn't the best.

crobbins215 commented:
Added a diagram description and also updated the diagram, worked with product design on improving the look of it following Camunda design guides.

akeller commented:
@christinaausley this is ready for final review

akeller commented:
@crobbins215 any other updates? Or should we merge and get this queued for a release?
